UnityProject/Assets/Scripts/UI/Systems/AdminTools/Interfaces/UITAGShowAndHiding.cs

```diff
@@ -0,0 +1,62 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using UnityEngine;
+
+public class UITAGShowAndHiding : MonoBehaviour
+{
+
+	public TAGType TAGType;
+
+	public List<TAGType> NeedsAny;
```

Comment:
instead of two lists, you could do what we do in clearance code. It is basically the same logic